docs/content/config/security/ssl.md

```diff
@@ -158,6 +158,81 @@ Certbot provisions certificates to `/etc/letsencrypt`. Add a volume to store the
 
     Certbot does support [alternative certificate providers via the `--server`][certbot::custom-ca] option. In most cases you'll want to use the default _Let's Encrypt_.
 
+#### Example using `certbot-dns-cloudflare` with Docker { data-toc-label='certbot-dns-cloudflare with Docker' }
+
+If for some reason, you can't expose your mail server's port 80 and port 443 (known as the `HTTP-01` challenge), you can consider obtaining a certificate by the `DNS-01` challenge.
+
+First, you need to obtain a Cloudflare API token
+
+1. Login into your Cloudflare dashboard.
+2. Navigate to MyProfile -> API Tokens.
+3. Create token a using the `Edit zone DNS` template since certbot [requires](https://certbot-dns-cloudflare.readthedocs.io/en/stable/#credentials) `ZONE:DNS:Edit` permission.
+
+!!! Zone resource configuration
+
+    Remember to choose a specific TLD, and create a token that allows access to all zone's resources is insecure here.
+
+4. Create `cloudflare.ini` with the following content:
+
+    ```dosini
+    dns_cloudflare_api_token = YOUR_CLOUDFLARE_TOKEN_HERE
+    ```
+
+5. Save it to the `.secrets` folder or anywhere you like. (Make sure to restrict its permission with the `chmod 600` command
+6. Add the following code into `docker-compose.yml`
+
+    ```yaml
+    services:
+      mailserver:
+        environments:
+          # Set SSL certificate type.
+          - SSL_TYPE=letsencrypt
+        volumes:
+          # We need to mount the cert generated by certbot into mail server.
+          - ./docker-data/certbot/certs/:/etc/letsencrypt
+      certbot-cloudflare:
+        image: certbot/dns-cloudflare:latest
+        container_name: certbot-cloudflare
+        volumes:
+          - ./docker-data/certbot/certs/:/etc/letsencrypt/
+          - ./docker-data/certbot/logs/:/var/log/letsencrypt/
+          - ./.secrets:/.secrets
+        command: certonly --dns-cloudflare --dns-cloudflare-credentials /.secrets/certbot/cloudflare.ini -d mail.example.com
+    ```
+
+    Alternative using the `docker run` command
+
+      ```sh
+      docker run \
+        --volume "${PWD}/docker-data/certbot/certs/:/etc/letsencrypt/" \
+        --volume "${PWD}/docker-data/certbot/logs/:/var/log/letsencrypt/" \
+        --volume "${PWD}/.secrets:/.secrets"
+        certbot/dns-cloudflare \
+        certonly --dns-cloudflare --dns-cloudflare-credentials /.secrets/certbot/cloudflare.ini -d mail.example.com
+      ```
+
+7. Turn it on to generate certificate.
+
+    ```sh
+    docker-compose up certbot-cloudflare
```

Comment:
Is this meant to run in the background to handle renewals at all? Or should the user manually invoke regularly?

If it's only intended for running manually `docker-compose run certbot-cloudflare` might be more suitable?

Reply:
Thanks for these suggestions, sorry I had a business trip yesterday.  
Generate certificate via certbot-cloudflare should be run manually since the part of renew certificate requires another action

Reply:
I've applied changes in [b32e713](https://github.com/docker-mailserver/docker-mailserver/pull/2968/commits/b32e71377769710093075156a04b2c0820b83b23), thanks for advice